core/embed/rust/src/micropython/map.rs

```diff
@@ -19,9 +19,10 @@ impl Map {
         // .all_keys_are_qstrs = 1,
         // .is_fixed = 1,
         // .is_ordered = 1,
+        // .no_realloc = 1,
         // .used = MP_ARRAY_SIZE(table_name),
         // .alloc = MP_ARRAY_SIZE(table_name),
-        let bits = 0b111 | N << 3;
+        let bits = 0b1111 | N << 4;
```

Comment:
could you please have a look at bindgen, to see whether this limitation was lifted?

Reply:
https://github.com/trezor/trezor-firmware/pull/4842/commits/24a328403df8add1071211f0d7a96458faa5784d doesn't require changing MicroPython code & its Rust bindings.